src/app/pages/RewardPage/components/ClaimForms/FeesEarnedClaimRow/index.tsx

```diff
@@ -55,42 +56,100 @@ export const FeesEarnedClaimRow: React.FC<IFeesEarnedClaimRowProps> = ({
 
   const { value: maxCheckpoints } = useCacheCallWithValue(
     'feeSharingProxy',
-    'numTokenCheckpoints',
+    'totalTokenCheckpoints',
     -1,
     contractAddress,
   );
-  const { send, ...tx } = useSendContractTx('feeSharingProxy', 'withdraw');
-  const { send: withdrawRBTC, ...tx2 } = useSendContractTx(
+
+  const {
+    userCheckpoint,
+    updateNextPositiveCheckpoint,
+  } = useGetNextPositiveCheckpoint(contractAddress, Number(maxCheckpoints));
+
+  const { send: withdraw, ...withdrawTx } = useSendContractTx(
+    'feeSharingProxy',
+    'withdraw',
+  );
+  const { send: withdrawRBTC, ...withdrawRBTCTx } = useSendContractTx(
     'feeSharingProxy',
     'withdrawRBTC',
   );
+  const {
+    send: withdrawStartingFromCheckpoint,
+    ...withdrawStartingFromCheckpointTx
+  } = useSendContractTx('feeSharingProxy', 'withdrawStartingFromCheckpoint');
+  const {
+    send: withdrawRBTCStartingFromCheckpoint,
+    ...withdrawRBTCStartingFromCheckpointTx
+  } = useSendContractTx(
+    'feeSharingProxy',
+    'withdrawRBTCStartingFromCheckpoint',
+  );
 
   const [isSimulating, setIsSimulating] = useState(false);
 
   const isClaimDisabled = useMemo(
     () =>
+      !userCheckpoint?.hasFees ||
       isSimulating ||
       claimFeesEarnedLocked ||
       !bignumber(amountToClaim).greaterThan(0) ||
       assetClaimLocked,
-    [isSimulating, claimFeesEarnedLocked, amountToClaim, assetClaimLocked],
+    [
+      userCheckpoint,
+      isSimulating,
+      claimFeesEarnedLocked,
+      amountToClaim,
+      assetClaimLocked,
+    ],
   );
 
   const onSubmit = useCallback(() => {
-    if (asset === Asset.RBTC) {
-      withdrawRBTC(
-        [0, address],
-        { from: address, gas: gasLimit[TxType.STAKING_REWARDS_CLAIM_RBTC] },
-        { type: TxType.STAKING_REWARDS_CLAIM_RBTC },
-      );
+    if (userCheckpoint?.hasSkippedCheckpoints) {
+      if (asset === Asset.RBTC) {
+        withdrawRBTCStartingFromCheckpoint(
+          [userCheckpoint?.checkpointNum, maxCheckpoints, address],
```

Comment:
I think that you can pass `maxCheckpoints` to all the withdrawing methods only if it's less than `300` (according to https://sovryn.atlassian.net/browse/SOV-2301?focusedCommentId=16960). I suggest creating a variable with `useMemo` for that.

Reply:
I don't know that so verify it with @tjcloa.

Reply:
To my understanding it should be capped to the 300 per transaction. So if maxCheckpoints > 300, you should use 300. Otherwise tx would fail as it would likely use more gas than block can handle. User will have to make second transaction to withdraw remaining checkpoints afterwards.